edx/edx-ora2
# EDUCATOR-5415: Add filename metadata to ORA submission files

**TL;DR -** When uploading ORA submission files, include a Content-Disposition header that specifies the file name, so when the file is downloaded it is downloaded with the "canonical" name of the file, rather than whatever default the backend provider uses. 

**What changed?**

- Added the one header to the file upload

**Note**

 This doesn't work for the django storage backend. This header changes nothing, the file is still downloaded as the random hex string filename. This is specifically for S3, which is what we use in prod. Even if it doesn't "work" with other backends, it won't break compatibility with any of them, it just may be ignored.

**Testing Instructions**

1. Create your own AWS account, and create a bucket in S3. Create a user with permissions for this bucket. Configure the bucket's CORS settings to accept anything from anywhere.
2. Navigate to or create `<edx-platform>/lms/envs/private.py`:
`ORA2_FILEUPLOAD_BACKEND = 's3'`
`FILE_UPLOAD_STORAGE_BUCKET_NAME = <your s3 bucket name>`
`AWS_ACCESS_KEY_ID = <your aws access key id>`
`AWS_SECRET_ACCESS_KEY = <your aws access key secret>`
`AWS_S3_ENDPOINT_URL = 'https://s3.us-east-1.amazonaws.com'` (as long as your bucket is in us-east-virginia)
3. Saving this file /should/ force the lms to restart, but if it doesn't, restart it.
4. Submit a file in an ORA. Click the resultant link. The file you download should have the file name of the file displayed on the page.
  
**Developer Checklist**

JIRA: [EDUCATOR-5415](https://openedx.atlassian.net/browse/EDUCATOR-5415)

FIY: @edx/masters-devs-gta
